File: LeetCode Practice Problems/119. Pascal's Triangle II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> ans;
        ans.push_back(1);
        for(int i=1;i<=rowIndex;i++)
        {
            vector<int> a;
            a.push_back(1);
            for(int j=0;j<ans.size()-1;j++)
            {
                a.push_back(ans[j] + ans[j+1]);
            }
            a.push_back(1);
            ans = a;
        }
        
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/119-Pascal's Triangle II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> ans = new ArrayList();
        ans.add(1);
        for(int i=1;i<=rowIndex;i++) {
            List<Integer> cur = new ArrayList();
            
            for(int j=0;j<ans.size() + 1;j++) {
                if(j == 0 || j == ans.size()) {
                    cur.add(1);
                } else {
                    cur.add(ans.get(j) + ans.get(j-1));
                }
            }
            ans = cur;
        }
        return ans;
    }
}
